src/middlewared/middlewared/gdb.py

```diff
@@ -0,0 +1,190 @@
+import argparse
+import logging
+import libzfs
+import os
+import re
+import requests
+import subprocess
+import tempfile
+import textwrap
+import tqdm
+
+
+class MiddlewareGDB(object):
+
+    def __init__(self):
+        self.logger = logging.getLogger('middlewaregdb')
+
+    def get_dataset(self, name=None):
+        zfs = libzfs.ZFS()
+        if name is None:
+            for pool in zfs.pools:
+                if pool.name == 'freenas-boot':
+                    continue
+                return pool.root_dataset
+            raise RuntimeError('no dataset found')
+        else:
+            return zfs.get_dataset(name)
+
+    def _get_version(self):
+        with open('/etc/version', 'r') as f:
+            version = f.read().split()[0]
+
+        product, version = version.split('-', 1)
+        major = version.split('-')[0].split('.')[0]
+
+        return {
+            'major': major,
+            'product': product,
+            'version': version,
+        }
+
+    def debug_symbols_url(self):
+        version = self._get_version()
+        if version['product'] == 'FreeNAS':
+            base_url = f'https://download.freenas.org/{version["major"]}'
+            if 'MASTER' in version['version']:
+                verdir = f'MASTER/{version["version"].split("-")[-1]}'
+            else:
+                verdir = version['version']
+            dirurl = f'{base_url}/{verdir}/x64/'
+            req = requests.get(dirurl)
+            if req.status_code != 200:
+                raise RuntimeError('debug symbols url could not be found, provide the url')
```

Comment:
Very confusing error message. Let's include HTTP response code here instead of `not found`